core/frontend/src/main/scala/io/udash/utils/FileUploader.scala

```diff
@@ -0,0 +1,77 @@
+package io.udash.utils
+
+import io.udash._
+import org.scalajs.dom.raw.{FormData, XMLHttpRequest}
+import org.scalajs.dom._
+
+import scala.concurrent.ExecutionContext
+import scala.scalajs.js
+
+class FileUploader(url: Url)(implicit ec: ExecutionContext) {
+  import FileUploader._
+
+  def upload(input: html.Input): ReadableModelProperty[FileUploadModel] = {
+    val builder = Seq.newBuilder[File]
+    val files = input.files
+    for (i <- 0 until files.length)
+      builder += files(i)
+
+    upload(input.name, builder.result())
+  }
+
+  def upload(fieldName: String, files: Seq[File],
+             extraData: Map[js.Any, js.Any] = Map.empty): ReadableModelProperty[FileUploadModel] = {
+    val p = ModelProperty[FileUploadModel]
+    val data = new FormData()
+
+    extraData.foreach(item => data.append(item._1, item._2))
+    files.foreach(file => {
+      data.append(s"$fieldName[]", file)
```

Comment:
Why is this "[ ]" required? Does this connect with FileUploadServlet:17 - it's stripped there.